test/integration_test/fafb_matrix_test.go

```diff
@@ -0,0 +1,305 @@
+	return func(t *testing.T) {
+		// Check that we have enough backends to support this test: skip early
+		fleetBackends := GenerateFleetOrSkip(t, pxNamespace, c.BackendRequirements)
+
+		// Get versions from URL
+		logrus.Infof("Get component images from versions URL")
+		imageListMap, err := testutil.GetImagesFromVersionURL(pxSpecGenURL)
+		require.NoError(t, err)
+
+		// Construct Portworx StorageCluster object
+		cluster, err := constructStorageCluster(pxSpecGenURL, imageListMap)
+		require.NoError(t, err)
+
+		c.PopulateStorageCluster(cluster)
+
+		// Add extra env variables
+		releaseManifestURL := pxSpecGenURL
+		if !strings.HasSuffix(releaseManifestURL, "/") {
+			releaseManifestURL += "/"
+		}
+		releaseManifestURL += "version"
+
+		cluster.Spec.Env = append(cluster.Spec.Env, []k8sv1.EnvVar{
+			{
+				Name:  "PX_RELEASE_MANIFEST_URL",
+				Value: releaseManifestURL,
+			},
+			{
+				Name:  "PX_LOGLEVEL",
+				Value: "debug",
+			},
+		}...)
+
+		// Create px-pure-secret
+		logrus.Infof("Create or update %s in %s", OutputSecretName, pxNamespace)
+		err = createPureSecret(fleetBackends, pxNamespace)
+		require.NoError(t, err)
+
+		// Deploy cluster
+		logrus.Infof("Create StorageCluster %s in %s", cluster.Name, cluster.Namespace)
+		cluster, err = createStorageCluster(cluster)
+		require.NoError(t, err)
+
+		// Validate cluster deployment
+		logrus.Infof("Validate StorageCluster %s", cluster.Name)
+		err = testutil.ValidateStorageCluster(imageListMap, cluster, defaultValidateDeployTimeout, defaultValidateDeployRetryInterval, c.ShouldStartSuccessfully, "")
+		require.NoError(t, err)
+
+		// Delete cluster
+		logrus.Infof("Delete StorageCluster %s", cluster.Name)
+		err = testutil.UninstallStorageCluster(cluster)
+		require.NoError(t, err)
+
+		// Validate cluster deletion
+		logrus.Infof("Validate StorageCluster %s deletion", cluster.Name)
+		err = testutil.ValidateUninstallStorageCluster(cluster, defaultValidateUninstallTimeout, defaultValidateUninstallRetryInterval)
+		require.NoError(t, err)
+
+		// Delete px-pure-secret
+		logrus.Infof("Delete Secret %s", OutputSecretName)
+		err = deletePureSecretIfExists(cluster.Namespace)
+		require.NoError(t, err)
+
+		// TODO: mark test completion using testrail IDs
+	}
+}
+
+func createPureSecret(config DiscoveryConfig, namespace string) error {
+	pureJSON, err := config.DumpJSON()
+	if err != nil {
+		return err
+	}
+
+	if err = deletePureSecretIfExists(namespace); err != nil {
+		return err
+	}
+
+	// Wait a little bit
+	time.Sleep(time.Second * 5)
+
+	_, err = coreops.Instance().CreateSecret(&k8sv1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      OutputSecretName,
+			Namespace: namespace,
+		},
+		Data: map[string][]byte{
+			"pure.json": pureJSON,
+		},
+	})
+	return err
+}
+
+func deletePureSecretIfExists(namespace string) error {
+	if err := coreops.Instance().DeleteSecret(OutputSecretName, namespace); !errors.IsNotFound(err) && !errors.IsAlreadyExists(err) {
```

Comment:
nit: No need to check `errors.IsAlreadyExists`. I don't think k8s apiserver will return that error for a Delete API. It would only happen for a Create API.